include/exadg/convection_diffusion/driver.cpp

```diff
@@ -101,8 +88,11 @@ Driver<dim, Number>::setup()
 
   matrix_free = std::make_shared<MatrixFree<dim, Number>>();
   if(application->get_parameters().use_cell_based_face_loops)
-    Categorization::do_cell_based_loops(*grid->triangulation, matrix_free_data->data);
-  matrix_free->reinit(*grid->get_dynamic_mapping(),
+    Categorization::do_cell_based_loops(*application->get_grid()->triangulation,
+                                        matrix_free_data->data);
+  std::shared_ptr<Mapping<dim> const> mapping =
+    get_dynamic_mapping<dim, Number>(application->get_grid(), grid_motion);
```

Comment:
Since you write the code like this, I assume the template arguments cannot be automatically deduced? If they can, I would prefer to use
```suggestion
    get_dynamic_mapping(application->get_grid(), grid_motion);
```
as the template arguments do not matter for what we want to achieve.

Reply:
Yes, the compiler says "template argument deduction/substitution failed", something that I do not really understand why it happens or why this information can not be deduced from the second argument `std::shared_ptr<GridMotionBase<dim, Number>> grid_motion;`.